clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -990,6 +990,10 @@ public Node leastLoadedNode(long now) {
                 newNodes.add(new Node(nodeId--, address.getHostString(), address.getPort()));
             }
 
+            if (newNodes.size() == 0) {
+                return null;
```

Comment:
Returning `null` appears to be safe as the caller has conditions around the call. The algorithms also return `null` when they fail. This change looks safe to me.